### ISSUE TITLE
brew install php@5.6symbol(s) not found for architecture x86_64

### ISSUE THREAD

**Issue by 0josh0** [eXolnet/homebrew-deprecated#9]
brew install php@5.6
Updating Homebrew...
==> Installing php@5.6 from exolnet/deprecated
==> Installing dependencies for exolnet/deprecated/php@5.6: openssl@1.1 and openssl
==> Installing exolnet/deprecated/php@5.6 dependency: openssl@1.1
==> Downloading https://homebrew.bintray.com/bottles/openssl@1.1-1.1.1d.high_sierra.bottle.tar.gz
Already downloaded: /Users/liangjianhua/Library/Caches/Homebrew/downloads/3e87a1132e64eed274a253d12489b47f7f8dd5350d00a036efeab0d443ea0728--openssl@1.1-1.1.1d.high_sierra.bottle.tar.gz
==> Pouring openssl@1.1-1.1.1d.high_sierra.bottle.tar.gz
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@1.1/certs

and run
  /usr/local/opt/openssl@1.1/bin/c_rehash

openssl@1.1 is keg-only, which means it was not symlinked into /usr/local,
because openssl/libressl is provided by macOS so don't link an incompatible version.

If you need to have openssl@1.1 first in your PATH run:
  echo 'export PATH="/usr/local/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl@1.1 you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl@1.1/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl@1.1/include"

For pkg-config to find openssl@1.1 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig"

==> Summary
🍺  /usr/local/Cellar/openssl@1.1/1.1.1d: 7,983 files, 18.0MB
==> Installing exolnet/deprecated/php@5.6 dependency: openssl
==> Downloading https://homebrew.bintray.com/bottles/openssl-1.0.2s.high_sierra.bottle.tar.gz
Already downloaded: /Users/liangjianhua/Library/Caches/Homebrew/downloads/c2564385473dadff8948954ab9d4c2077d75f7f6c746114ff3fd35471ee51013--openssl-1.0.2s.high_sierra.bottle.tar.gz
==> Pouring openssl-1.0.2s.high_sierra.bottle.tar.gz
==> Caveats
A CA file has been bootstrapped using certificates from the SystemRoots
keychain. To add additional certificates (e.g. the certificates added in
the System keychain), place .pem files in
  /usr/local/etc/openssl/certs

and run
  /usr/local/opt/openssl/bin/c_rehash

openssl is keg-only, which means it was not symlinked into /usr/local,
because Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries.

If you need to have openssl first in your PATH run:
  echo 'export PATH="/usr/local/opt/openssl/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl/include"

For pkg-config to find openssl you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"

==> Summary
🍺  /usr/local/Cellar/openssl/1.0.2s: 1,795 files, 12.1MB
==> Installing exolnet/deprecated/php@5.6
==> Downloading https://php.net/get/php-5.6.40.tar.xz/from/this/mirror
Already downloaded: /Users/liangjianhua/Library/Caches/Homebrew/downloads/ab2ea146f7d8c5258e6d33e9522dac5528cbc3e966bc6e5689fa57860dbb5441--php-5.6.40.tar.xz
==> Patching
patching file acinclude.m4
Hunk #1 succeeded at 444 (offset 3 lines).
Hunk #2 succeeded at 459 (offset 3 lines).
Hunk #3 succeeded at 494 (offset 3 lines).
Hunk #4 succeeded at 506 (offset 3 lines).
Hunk #5 succeeded at 2500 with fuzz 2 (offset 81 lines).
patching file Zend/zend_compile.h
patching file Zend/zend_execute.h
==> ./buildconf --force
==> ./configure --prefix=/usr/local/Cellar/php@5.6/5.6.40 --localstatedir=/usr/local/var --sysconfdir=/usr/local/etc/php/5.6 --with-config-file-path=/usr/local/etc/php/5.6 --with-config-file-scan-dir=/usr
==> make
Last 15 lines from /Users/liangjianhua/Library/Logs/Homebrew/php@5.6/03.make:
      _load_all_certs_from_file in openssl.o
      _php_array_to_X509_sk in openssl.o
  "_sk_shift", referenced from:
      _load_all_certs_from_file in openssl.o
  "_sk_value", referenced from:
      _zif_openssl_x509_parse in openssl.o
      _zif_openssl_csr_new in openssl.o
      _zif_openssl_pkcs7_verify in openssl.o
      _php_openssl_parse_config in openssl.o
      _php_openssl_sockop_set_option in xp_ssl.o
      _capture_peer_certs in xp_ssl.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libs/libphp5.bundle] Error 1
make: *** Waiting for unfinished jobs....

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/exolnet/homebrew-deprecated/issues

**Comment by mihan007** [comment]
got same error

**Comment by pjblasco74** [comment]
I have same error.

**Comment by apogeo** [comment]
same problem here

**Comment by aas395** [comment]
Same issue for me

**Comment by luyenlvVNC** [comment]
My logs

brew install exolnet/deprecated/php@5.6
==> Installing php@5.6 from exolnet/deprecated
==> Downloading https://php.net/get/php-5.6.40.tar.xz/from/this/mirror
==> Downloading from https://www.php.net/distributions/php-5.6.40.tar.xz
                        -=O=-                         #    #    #   #         
==> Patching
patching file acinclude.m4
Hunk #1 succeeded at 444 (offset 3 lines).
Hunk #2 succeeded at 459 (offset 3 lines).
Hunk #3 succeeded at 494 (offset 3 lines).
Hunk #4 succeeded at 506 (offset 3 lines).
Hunk #5 succeeded at 2500 with fuzz 2 (offset 81 lines).
patching file Zend/zend_compile.h
patching file Zend/zend_execute.h
==> ./buildconf --force
==> ./configure --prefix=/usr/local/Cellar/php@5.6/5.6.40 --localstatedir=/usr/local/var --sysconfdir=/usr/local/e
==> make
Last 15 lines from /Users/levanluyen/Library/Logs/Homebrew/php@5.6/03.make:
      _load_all_certs_from_file in openssl.o
  "_sk_value", referenced from:
      _zif_openssl_x509_parse in openssl.o
      _zif_openssl_csr_new in openssl.o
      _zif_openssl_pkcs7_verify in openssl.o
      _php_openssl_parse_config in openssl.o
      _php_openssl_sockop_set_option in xp_ssl.o
      _capture_peer_certs in xp_ssl.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libs/libphp5.bundle] Error 1
make: *** Waiting for unfinished jobs....
warning: no debug symbols in executable (-arch x86_64)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/exolnet/homebrew-deprecated/issues

**Comment by nanoguo** [comment]
> My logs
> 
> brew install exolnet/deprecated/php@5.6
> ==> Installing php@5.6 from exolnet/deprecated
> ==> Downloading https://php.net/get/php-5.6.40.tar.xz/from/this/mirror
> ==> Downloading from https://www.php.net/distributions/php-5.6.40.tar.xz
> -=O=- # # # #
> ==> Patching
> patching file acinclude.m4
> Hunk #1 succeeded at 444 (offset 3 lines).
> Hunk #2 succeeded at 459 (offset 3 lines).
> Hunk #3 succeeded at 494 (offset 3 lines).
> Hunk #4 succeeded at 506 (offset 3 lines).
> Hunk #5 succeeded at 2500 with fuzz 2 (offset 81 lines).
> patching file Zend/zend_compile.h
> patching file Zend/zend_execute.h
> ==> ./buildconf --force
> ==> ./configure --prefix=/usr/local/Cellar/php@5.6/5.6.40 --localstatedir=/usr/local/var --sysconfdir=/usr/local/e
> ==> make
> Last 15 lines from /Users/levanluyen/Library/Logs/Homebrew/php@5.6/03.make:
> _load_all_certs_from_file in openssl.o
> "_sk_value", referenced from:
> _zif_openssl_x509_parse in openssl.o
> _zif_openssl_csr_new in openssl.o
> _zif_openssl_pkcs7_verify in openssl.o
> _php_openssl_parse_config in openssl.o
> _php_openssl_sockop_set_option in xp_ssl.o
> _capture_peer_certs in xp_ssl.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make: *** [libs/libphp5.bundle] Error 1
> make: *** Waiting for unfinished jobs....
> warning: no debug symbols in executable (-arch x86_64)
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols
> 
> If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
> https://github.com/exolnet/homebrew-deprecated/issues

same as yours

**Comment by orlandoloto** [comment]
help!

**Comment by zerokeeper** [comment]
I solved this problem,it's openssl problem.Don't use `openssl@1.1`,use openssl.
Switch openssl to `1.0.2s` or other version,it will install successfully.
```
export PATH="/usr/local/opt/openssl/bin:$PATH"
export LDFLAGS="-L/usr/local/opt/openssl/lib"
export CPPFLAGS="-I/usr/local/opt/openssl/include"
```
then `brew install php@5.6`,it will success install.

**Comment by thibaudpauwels** [comment]
I am having the same issue as the people above 

Updating Homebrew...
==> Installing php@5.6 from exolnet/deprecated
==> Downloading https://php.net/get/php-5.6.40.tar.xz/from/this/mirror
==> Patching
patching file acinclude.m4
Hunk #1 succeeded at 444 (offset 3 lines).
Hunk #2 succeeded at 459 (offset 3 lines).
Hunk #3 succeeded at 494 (offset 3 lines).
Hunk #4 succeeded at 506 (offset 3 lines).
Hunk #5 succeeded at 2500 with fuzz 2 (offset 81 lines).
patching file Zend/zend_compile.h
patching file Zend/zend_execute.h
==> ./buildconf --force
==> ./configure --prefix=/usr/local/Cellar/php@5.6/5.6.40 --localstatedir=/usr/local/var --sysconfdir=/usr/local/etc/php/5.6 --with-config-file-path=/usr/local
==> make
Last 15 lines from /Users/tpauwels/Library/Logs/Homebrew/php@5.6/03.make:
      _load_all_certs_from_file in openssl.o
  "_sk_value", referenced from:
      _zif_openssl_x509_parse in openssl.o
      _zif_openssl_csr_new in openssl.o
      _zif_openssl_pkcs7_verify in openssl.o
      _php_openssl_parse_config in openssl.o
      _php_openssl_sockop_set_option in xp_ssl.o
      _capture_peer_certs in xp_ssl.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libs/libphp5.bundle] Error 1
make: *** Waiting for unfinished jobs....
warning: no debug symbols in executable (-arch x86_64)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ext/opcache/.libs/opcache.a(shared_alloc_shm.o) has no symbols

**Comment by pgrenaud** [comment]
Hi!

According to [this PHP bug](https://bugs.php.net/bug.php?id=73741), `PHP 5.6` was never meant to be built with `OpenSSL 1.1.0`. The right version is `OpenSSL 1.0.2s` as indicated by the dependencies.

According to the logs, @zerokeeper is right. You have to manually set your envvars in order to let PHP find the right version of OpenSSL. This might be different from system to system, so check the output of `brew info openssl`.

Right now it seems that you have to do the following:

    PATH="/usr/local/opt/openssl/bin:$PATH"
    LDFLAGS="-L/usr/local/opt/openssl/lib"
    CPPFLAGS="-I/usr/local/opt/openssl/include"
    PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"

Then run `brew install php@5.6` again.

I also encourage you all to update your system to the latest version of macOS (as of today, Mojave 10.14.6) and to upgrade your code to the latest version of PHP (as of today, 7.3.9).

**Comment by thibaudpauwels** [comment]
I am still having the issue above with openssl 1.0.2s and doing the instruction above. 
Still getting the "ld: symbol(s) not found for architecture x86_64" error.

I need this version of php to run a legacy piece of software.
Any ideas what could be causing this issue ?

Kind regards,
Thibaud

**Comment by pgrenaud** [comment]
@thibaudpauwels At this point, we are not able to provide any further help. Sorry about that.

**Comment by Martin-B** [comment]
I can confirm that upgrading the OS to 10.14+ (from High Sierra) and upgrading Xcode and XCode compiler tools fixed my issue with reinstalling php 5.6, vital for our maintenance of a large number Drupal 7 websites. Thanks again for providing this tap and migrating the legacy formulas.

**Comment by pgrenaud** [comment]
@Martin-B Awesome! Glad to hear that! Thanks for letting us know.
